api/routes/validation-test.ts

```diff
@@ -0,0 +1,61 @@
+import { Type } from '@sinclair/typebox';
+
+import type { FastifyInstanceWithTypeProvider } from '..';
```

Comment:
How does this not create an infinite import loop thing?

Reply:
`import type` is handled differently from `import`.

I guess it works like this: TS imports a file, creating placeholders for the imported types, then it goes to the second file and does the same. At that point it can fill in any placeholders with the imported types and it can do so for both files. 

Given that TS is totally fine with circular types, I can't see there ever being a problem.

Reply:
Sure. If this is not a problem with TS things, then I am happy with this.

Reply:
One nice thing is that the circular dependency plugin can tell the difference between the two.